pkg/keymanagementprovider/refresh/kubeRefresh.go

```diff
@@ -151,62 +94,36 @@ func (kr *KubeRefresher) GetResult() interface{} {
 	return kr.Result
 }
 
+func (kr *KubeRefresher) GetStatus() interface{} {
+	return kr.Status
+}
+
 // Create creates a new KubeRefresher instance
 func (kr *KubeRefresher) Create(config map[string]interface{}) (Refresher, error) {
-	client, ok := config["client"].(client.Client)
+	provider, ok := config["provider"].(kmp.KeyManagementProvider)
```

Comment:
Usually we need to check exist first and then do type assertion to avoid panic. But if we make config a struct instead of map, we could fetch it via config.Provider.